lookout/core/lib.py

```diff
@@ -86,21 +91,57 @@ def files_by_language(files: Iterable[File]) -> Dict[str, Dict[str, File]]:
     return result
 
 
-def filter_files(files: Dict[str, File], line_length_limit: int,
-                 log: logging.Logger = None) -> List[File]:
+def filter_filepaths(filepaths: Iterable[str], exclude_pattern: Optional[str] = None,
+                     ) -> Iterable[str]:
     """
-    Filter files based on their maximum line length.
+    Mirror of the file filtering used in the format analyzer for use by debugging tools.
 
-    :param files: files to filter.
+    :param filepaths: Iterable of filepaths to filter.
+    :param exclude_pattern: Pattern to reject files based on their path. If None, uses the pattern
+                            currently in use in lookout.core. Use "" to not filter anything.
+    :return Iterable of paths, filtered.
+    """
+    if exclude_pattern is None:
+        exclude_pattern = GARBAGE_PATTERN
+    exclude_compiled_pattern = re.compile(exclude_pattern) if exclude_pattern else None
+    for filepath in filepaths:
+        if not isfile(filepath):
+            continue
+        if exclude_compiled_pattern and exclude_compiled_pattern.search(filepath):
+            continue
+        yield filepath
+
+
+def filter_files(filenames: Iterable[str], line_length_limit: int, client: BblfshClient,
+                 language: str, log: logging.Logger = None) -> Iterable[File]:
```

Comment:
We return `Iterator`, not `Iterable`.

cc/ @m09 - this is what I was talking about.

Reply:
Yes I'll create an issue because we have this problem all over the codebase